### ISSUE TITLE
Different animation for page enter and page leave will break the back navigation

### ISSUE THREAD

**Issue by NoNameProvided** [driftyco/ionic#8436]
#### Short description of the problem:

When a different animation is used for navigating back than the one was used to push the current page, the animation stuck halfway. 

**Edit:** Most of the time this only happens if I try to pop from the entered page, if I navigate to a third page and pop back using a custom transition, everything works. 

#### What behavior are you expecting?

I can push and pop with any custom transition, and the two is not tied to each other.

**Steps to reproduce:**
1. Push a page with iOS transition
2. Pop the page with android transition
3. See that the back animation get stucked, half of the screen is black, but the visible part of the parent page is usable. 
 
**Which Ionic Version?** 2.x

**Run `ionic info` from terminal/cmd prompt:**

```
Cordova CLI: 6.3.1
Gulp version:  CLI version 1.2.2
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-rc.0
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: 1.9.0 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v6.3.1
Xcode version: Xcode 8.0 Build version 8A218a 
```

**Comment by manucorporat** [comment]
Duplicate of #7171

Will be fixed once #8272 is merged!

**Comment by NoNameProvided** [comment]
Cool, thanks!
